src/shared-components/carousel/index.tsx

```diff
@@ -45,195 +46,140 @@ type CarouselProps = {
   numCardsVisible: 1 | 2 | 3;
 };
 
-type CarouselState = {
-  currentIndex: number;
-  lastIndex: number;
-};
-
 /**
  * Carousels should be used to provide valuable information or additional context on a page. One of the best examples of a Carousel is for product recommendations.
  *
  * The property `numCardsVisible` will define the width of the Carousel Container.
  *
  * An array of `Carousel.Card` must be used for the carousel content. It includes the base styles for the Card which may be extended as shown above.
  */
-export class Carousel extends React.Component<CarouselProps, CarouselState> {
-  static Card = Card;
-
-  static propTypes = {
-    autoplay: PropTypes.bool,
-    autoplaySpeed: PropTypes.number,
-    carouselType: PropTypes.oneOf(['primary', 'secondary']),
-    centerMode: PropTypes.bool,
-    /**
-     * Array of `Carousel.Card`
-     */
-    children: PropTypes.arrayOf(PropTypes.node).isRequired,
-    hideArrows: PropTypes.bool,
-    hideDots: PropTypes.bool,
-    bottomRightAlignedArrows: PropTypes.bool,
-    infinite: PropTypes.bool,
-    numCardsVisible: PropTypes.number.isRequired,
-  };
-
-  static defaultProps = {
-    autoplay: false,
-    autoplaySpeed: 5000,
-    carouselType: 'primary',
-    centerMode: true,
-    hideArrows: false,
-    hideDots: false,
-    bottomRightAlignedArrows: false,
-    infinite: false,
+export const Carousel = ({
+  autoplay = false,
+  autoplaySpeed = 5000,
+  bottomRightAlignedArrows = false,
+  carouselType = 'primary',
+  centerMode = true,
+  children,
+  hideArrows = false,
+  hideDots = false,
+  infinite = false,
+  numCardsVisible,
+}: CarouselProps) => {
+  const getLastIndex = () => {
+    const numberSlides = children.length;
+    return numberSlides - numCardsVisible;
   };
 
-  slider: React.RefObject<Slider> = React.createRef();
-
-  timeoutId: number | null = null;
-
-  userHasInteracted = false;
-
-  constructor(props: CarouselProps) {
-    super(props);
-
-    this.state = {
-      currentIndex: 0,
-      lastIndex: this.getLastIndex(),
-    };
-  }
-
-  componentDidUpdate() {
-    if (this.shouldReplay) {
-      this.replay();
-    }
-
-    this.updateLastIndex();
-  }
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [lastIndex, setLastIndex] = useState(getLastIndex());
```

Comment:
```suggestion
  const [lastIndex, setLastIndex] = useState(getLastIndex);
```

It's minor, but `useState` also accepts a function so that we only invoke the method once, instead of every time the component re-renders: https://www.benmvp.com/blog/four-characters-optimize-react-component/

Reply:
fixed here https://github.com/curology/radiance-ui/pull/461/commits/95492678284de09c1f4baf05ab2b951aa5be6d2a